### PR TITLE
Issue 157: "Used recently" tab

### PR DESCRIPTION
**Pull request by artkravchenko**
Matched to `feature/recently-used-tags`.
Card: https://trello.com/c/cVPwiIda/338-issue-157-used-recently-tab-for-location-tags-hashtags-school-tags

### CODE REVIEW

File: src/components/add-tag-modal/add-hashtag-form.js [hunk 1]
```diff
@@ -16,20 +16,62 @@
...
+  async getRecentHashtags() {
+    const client = new ApiClient(API_HOST);
+    try {
+      const hashtags = await client.userRecentHashtags();
+      this.setState({ recentHashtags: hashtags });
+      return hashtags;
+    } catch (e) {
+      return e.message;
+    }
+  }
```
Comment (indeyets) [inline src/components/add-tag-modal/add-hashtag-form.js]: this should be done via redux

File: src/components/add-tag-modal/add-hashtag-form.js [hunk 2]
```diff
@@ -55,32 +106,62 @@ export default class AddHashtagForm extends Component {
   };
 
   render() {
+    let recentHashtags = [];
+    if (Array.isArray(this.state.recentHashtags)) {
+      recentHashtags = _.clone(this.state.recentHashtags).filter((tag, i) => this.state.cut.indexOf(i) === -1);
+    }
+
+    const popularHashtags = ['first', 'second', 'third'];
```
Comment (indeyets) [inline src/components/add-tag-modal/add-hashtag-form.js]: what's this? some kind of temporary data? shouldn't be in real component

File: src/components/add-tag-modal/add-hashtag-form.js [hunk 3]
```diff
@@ -16,20 +16,62 @@
...
-    })).isRequired,
     onAddHashtag: PropTypes.func.isRequired
   };
 
+  constructor(props) {
+    super(props);
+    this.addedHashtags = [];
+    this.state = {
+      recentHashtags: [],
+      cut: []
```
Comment (indeyets) [inline src/components/add-tag-modal/add-hashtag-form.js]: what does `cut` stand for in this component? selectedHashtags?

Comment (Oopscurity) [inline src/components/add-tag-modal/add-hashtag-form.js]: Yes. So they're cut from the cloud.

Comment (indeyets) [inline src/components/add-tag-modal/add-hashtag-form.js]: so, rename it to `selectedHashtags` then. it's way easier to understand

File: src/components/add-tag-modal/add-hashtag-form.js (hunk 2 above)
Comment (Oopscurity) [inline src/components/add-tag-modal/add-hashtag-form.js]: It's already been removed from here.

File: src/components/add-tag-modal/add-hashtag-form.js (hunk 1 above)
Comment (Oopscurity) [inline src/components/add-tag-modal/add-hashtag-form.js]: How should I rewrite this using Redux?

**Comment by indeyets** [comment]
this one is merged manually into `feature/tabs-for-tags` branch

**Pull request closed without merging**